File: Seccion 12 - Punteros en C++/Seccion_12_Python/07_ejercicio_07.py

nombre_usuario: str = ''

def pedir_nombre():
    global nombre_usuario
    
    nombre_usuario = input('Digite su nombre: ')
    nombre_usuario = nombre_usuario.upper()

def contarVocales(nombre: str) -> int:
    
    cont: int = 0
    i: int = 0
    
    VOCALES: list = ['A', 'E', 'I', 'O', 'U']
    
    while (i < len(nombre)):
        if nombre[i] in VOCALES:
            cont += 1
        
        i += 1
    
    return cont

def main():
    pedir_nombre()
    
    print('\nEl número de vocales en el nombre es: {}'.format(contarVocales(nombre_usuario)))

if __name__ == '__main__':
    main()

File: Seccion 12 - Punteros en C++/Seccion_12_Dart/07_ejercicio_07.dart

import 'dart:io';

String nombreUsuario;

void pedirNombre() {

  stdout.write('Digite su nombre: ');
  nombreUsuario = stdin.readLineSync().toUpperCase();

}

int contarVocales(String nombre) {
  print(nombre);
  int cont = 0, i = 0;

  while (i < nombre.length) {
    switch (nombre[i]) {
      case 'A':
      case 'E':
      case 'I':
      case 'O':
      case 'U':
        cont++;
    }

    i++;
  }

  return cont;
}

void main() {

  pedirNombre();

  print('\nEl número de vocales en el nombre es: ${contarVocales(nombreUsuario)}');

}